contracts/margined_engine/src/testing/position_liquidation_tests.rs

```diff
@@ -851,3 +959,205 @@ fn test_cannot_open_position_even_thought_short_is_underwater_if_still_under_aft
+        )
+        .unwrap();
+    router.execute(alice.clone(), msg).unwrap();
+
+    router.update_block(|block| {
+        block.time = block.time.plus_seconds(15);
+        block.height += 1;
+    });
+
+    let msg = vamm.set_spread_ratio(Uint128::from(1_000_000u128)).unwrap(); // 0.001
+    router.execute(owner.clone(), msg).unwrap();
+    let msg = vamm
+        .set_fluctuation_limit_ratio(Uint128::from(562_400_000u128))
+        .unwrap(); // 0.5624
+    router.execute(owner.clone(), msg).unwrap();
+
+    // the price will be dropped to 10 if we close whole position
+    // the price fluctuation will be (10 - 6.4) / 6.4 = 0.5625
+    // only 25% position (25 * 0.25 = 6.25) will be closed,
+    // position notional is 42.11
+    // amm reserves after 842.11 : 118.75
+    let msg = engine
+        .close_position(vamm.addr().to_string(), Uint128::zero())
+        .unwrap();
+    router.execute(alice.clone(), msg).unwrap();
+
+    let position = engine
+        .position(&router, vamm.addr().to_string(), alice.to_string())
+        .unwrap();
+    assert_eq!(position.size, Integer::new_negative(18_750_000_000u128));
+    assert_eq!(position.margin, Uint128::from(20_000_000_000u128));
+
+    // 5000 - open pos margin (25) + fee (-73.53 * 0.1%)
+    let alice_balance = usdc.balance::<_, _, Empty>(&router, alice.clone()).unwrap();
+    assert_eq!(alice_balance, Uint128::from(4_979_957_894_737u128));
+}
+
+#[test]
+fn test_close_whole_partial_position_when_partial_liquidation_ratio_is_one() {
+    let SimpleScenario {
+        mut router,
+        owner,
+        alice,
+        engine,
+        vamm,
+        ..
+    } = SimpleScenario::new();
+
+    let msg = engine
+        .set_partial_liquidation_ratio(Uint128::from(1_000_000_000u128))
+        .unwrap();
+    router.execute(owner.clone(), msg).unwrap();
+
+    router.update_block(|block| {
+        block.time = block.time.plus_seconds(15);
+        block.height += 1;
+    });
+
+    let msg = engine
+        .open_position(
+            vamm.addr().to_string(),
+            Side::Buy,
+            to_decimals(25u64),
+            to_decimals(10u64),
+            to_decimals(0u64),
+            vec![],
+        )
+        .unwrap();
+    router.execute(alice.clone(), msg).unwrap();
+
+    router.update_block(|block| {
+        block.time = block.time.plus_seconds(15);
+        block.height += 1;
+    });
+
+    let msg = vamm.set_spread_ratio(Uint128::from(1_000_000u128)).unwrap(); // 0.001
+    router.execute(owner.clone(), msg).unwrap();
+    let msg = vamm
+        .set_fluctuation_limit_ratio(Uint128::from(359_000_000u128))
+        .unwrap(); // 0.5624
+    router.execute(owner.clone(), msg).unwrap();
+
+    // the price will be dropped to 10 if we close whole position
+    // the price fluctuation will be (10 - 6.4) / 6.4 = 0.5625
+    // only 25% position (25 * 0.25 = 6.25) will be closed,
+    // position notional is 42.11
+    // amm reserves after 842.11 : 118.75
```

Comment:
comment is from `close_short_position` test and not `close_long_position` test, so the numbers arent the same